[full-ci] Upgrade to Symfony5

phil-davis opened the pull request:
## Description
WIP - core is updating from Symfony 4 to 5.

This app uses Symfony and still says that it supports ownCloud 9. That is to allow people to export from old ownCloud server system. The existing app release 0.2.1 can still be available for that.

But we can update the dependencies and code to use the newer Symfony 5 LTS version. This should work with existing recent oC10 core releases, and will work with the upcoming core 10.13 release.

## How Has This Been Tested?
CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

phil-davis commented on lib/Serializer.php:
```diff
@@ -69,8 +70,4 @@ public function serialize($data) {
 		return $this->serializer->deserialize($data, $type, 'json', []);
 	}
-
-	public function setIgnoredAttributes($attributes = []) {
```
`setIgnoredAttributes` is no longer part of Symfony 5.

The new way is to use `AbstractNormalizer::IGNORED_ATTRIBUTES` - see code changes in `writelnToStream` below.

phil-davis commented:
@pako81 @jnweiger and whoever makes these decisions.

We could update this app to use Symfony 5 (and drop support for ownCloud 9, and for early ownCloud 10.*). Please decide if we do that.

The CI is passing here, and there are end-to-end acceptance test scenarios, so I think that the external behavior is preserved and working.

pako81 commented:
This seems to be a "special case" since this app is not a Marketplace app, so not among the ones we do officially support. Still I guess we should try to update it to use Symfony version 5.

jnweiger commented:
I'd also advocate to move forward. The version number is at 0.2.1, as per semver with a major version of 0, we can break compatibility when bumping the minor version.

Let's do this as 0.3.0 with an info.xml with core minver 10.12 - 
That should block installation on 9.x and early 10.x

DeepDiver1975 commented:
Was this app ever used anywhere? Has it still a need? @IljaN might know ....

jvillafanez commented:
I'm not sure about the state of the app, but compatibility will be important.
We need to app to export and then import the data, so it's easier if the app is the same in both servers. I think it's better to keep the requirements as low as possible and check specific versions if we run into incompatibilities. Hopefully, the amount of incompatibilities we need to deal with are few, so it should be fine as long as they're located and documented.

phil-davis commented:
This app has its own Symfony3 set of dependencies. It has been working alongside core with Symfony4, and I think that it will keep working with core having Symfony5.

I will leave this PR here. There is no hurry to decide - and maybe we can leave this app the way it is (with Symfony3). It is only used as a tool for migrating stuff, not a real-time part of production installations.

jnweiger reviewed (approved):
I'd like to merge that into the release-0.3.0 branch after v0.2.2  is released.
I'l also bump php minversion in the release branch and adjust changelogs etc...

phil-davis commented:
I rebased to resolve conflicts.
And added the last commit to get the PHP dependencies right up-to-date:
```
Updating dependencies
Lock file operations: 0 installs, 6 updates, 0 removals
  - Upgrading doctrine/deprecations (v1.0.0 => v1.1.1)
  - Upgrading phpdocumentor/type-resolver (1.7.1 => 1.7.2)
  - Upgrading phpstan/phpdoc-parser (1.21.0 => 1.22.1)
  - Upgrading symfony/filesystem (v5.4.23 => v5.4.25)
  - Upgrading symfony/property-info (v5.4.23 => v5.4.24)
  - Upgrading symfony/serializer (v5.4.23 => v5.4.25)
Writing lock file
```

phil-davis commented:
The `release-0.3.0` branch is too old. It does not work with the current drone CI:
https://drone.owncloud.com/owncloud/data_exporter/2594/2/1
cancel-previous-builds should not be in CI any more.
The release branch needs to be rebased to current master (or something done).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/data_exporter?pullRequest=222) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/data_exporter?pullRequest=222) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: phil-davis<br/>:x: jnweiger<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/data_exporter?pullRequest=222) it.</sub>

phil-davis reviewed (commented):
This looks good now. I am the PR originator, so I can't actually approve.
Feel free to merge to `release-0.3.0` when you like.
